book.rb

```diff
@@ -0,0 +1,10 @@
+# define Book class
+class Book
+  attr_accessor :title, :author
+  attr_reader :rental
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+  end
+end
```

Comment:
Great job on implementing the `Book` class! It looks like you've made progress towards meeting the project requirements for creating a many-to-many relationship between `Person` and `Book` using the intermediate class `Rental`. Your code seems to be on the right track, but there are a couple of points to consider for further improvement and alignment with the specified requirements:

1. **Relationship Setup:**
   Your `Book` class includes the necessary attributes `title`, `author`, and `rental`. However, to fully establish the many-to-many relationship, it's essential to maintain a collection of rentals associated with each book. In your `initialize` method, consider initializing the `@rentals` instance variable as an empty array to store rentals related to this book.

2. **Adding Rentals:**
   The requirement mentions that there should be a method for adding rentals to a book. You can create a method like `add_rental(person, date)` in your `Book` class, where you use the `Rental` class to create a new rental instance and add it to the `@rentals` array. This way, you establish the 'has-many' association between books and rentals.

Here's an example of how you could implement the `add_rental` method:
```ruby
class Book
  # ... (other attributes)

  def initialize(title, author)
    # ... (initialize other attributes)
    @rentals = []  # Initialize rentals as an empty array
  end

  def add_rental(person, date)
    rental = Rental.new(date, self, person)
    @rentals << rental  # Add the new rental to the rentals array
  end
end
```

By making these adjustments, your code will effectively create and manage the many-to-many relationship between `Person` and `Book` through the `Rental` intermediate class, as required.

Remember, you're making good progress! Feel free to refer to the provided resources for additional guidance on implementing the necessary methods and relationships. Keep up the good work!

Best regards,
Temitope
Code Reviewer
01/09/2023

---